### ISSUE TITLE
Following tests failed

### ISSUE THREAD

**Issue by diveshuttam** [dashohoxha/egpg#56]
These test fail on bionic container (branch gnupg2.2)
refer EasyGnuPG/pgpg#9
- [x] t27 (due to haveged not started)
- [x] t28 (due to haveged not started)
- [x] t73 (Dongle directory does not exist: /host/egpg/tests/trash directory.t73-ext-key2dongle/dongle/test1)
- [x] t31 (/host/egpg/tests/trash directory.t31-key-split/dongle/test1)
- [x] t51 (see below mostly seems some problem in key format at first look)

```
t51
not ok 8 - egpg contact ls -c
#
#           [[ $(egpg contact ls -c | grep fpr) == "fpr:::::::::A9446F790F9BE7C9D108FC6718D1DA4D9E7A4FD0:" ]] &&
#           [[ $(egpg contact ls --colons | grep fpr) == "fpr:::::::::A9446F790F9BE7C9D108FC6718D1DA4D9E7A4FD0:" ]]
#
```

**Comment by dashohoxha** [comment]
t73 fails on the last line.
It seems that root can write on a directory even if it is nor writable, so the test fails to detect a failure in the last line. Maybe we should remove this case from automatic testing, or should improve automatic testing to run as a normal (non-root) user.

**Comment by dashohoxha** [comment]
t31 fails for the same reason as t73

**Comment by dashohoxha** [comment]
t51 fails for the same reason that key revoke failed. You can check this by adding `gpg contact ls -c` on the last test case (and run the test with option `-d`).

**Comment by diveshuttam** [comment]
>t73 fails on the last line.
It seems that root can write on a directory even if it is nor writable, so the test fails to detect a failure in the last line. Maybe we should remove this case from automatic testing, or should improve automatic testing to run as a normal (non-root) user.

Yes, I tried it and it is the case. I think creating another user will be better and will be more close to real environments. I did look up for other approaches like `chattr +i` to make directory immutable; running the particular subtest as user `nobody` but they didn't work out. I'll add them after completing other parts mostly before second eval. For now I am removing this test.

> t51 fails for the same reason that key revoke failed. You can check this by adding gpg contact ls -c on the last test case (and run the test with option -d).

Checked it out, yup its because of the same reason.

Will fix them in a while.
